src/query/graphite/graphite/timespec.go

```diff
@@ -108,9 +137,146 @@ func ParseTime(s string, now time.Time, absoluteOffset time.Duration) (time.Time
 		return time.Unix(n, 0).UTC(), nil
 	}
 
+	ref, offset := s, ""
+	if strings.Contains(s, "+") {
+		stringSlice := strings.Split(s, "+")
+		ref, offset = stringSlice[0], stringSlice[1]
+		offset = "+" + offset
+	} else if strings.Contains(s, "-") {
+		stringSlice := strings.Split(s, "-")
+		ref, offset = stringSlice[0], stringSlice[1]
+		offset = "-" + offset
+	}
+	err = nil
+	parsedReference, err := ParseTimeReference(ref, now)
+	parsedOffset, err := ParseOffset(offset)
+	if err == nil {
+		return parsedReference.Add(parsedOffset), err
+	}
+
 	return now, err
 }
 
+
+func ParseTimeReference(ref string, now time.Time) (time.Time, error) {
+	hour := now.Hour()
+	minute := now.Minute()
+	refDate := time.Time{}
```

Comment:
Put these into a var block (so you don't need to explicitly declare refDate as zero value):
```go
var (
  hour = now.Hour()
  minute = now.Minute()
  refDate time.Time
)
```